experimental/packages/opentelemetry-sdk-metrics-base/src/exemplar/SimpleFixedSizeExemplarReservoir.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Context, HrTime } from '@opentelemetry/api';
+import { ValueType, Attributes } from '@opentelemetry/api-metrics-wip';
+import { FixedSizeExemplarReservoirBase } from './ExemplarReservoir';
+
+/**
+ * Fixed size reservoir that uses equivalent of naive reservoir sampling
+ * algorithm to accept measurements.
+ * 
+ */
+export class SimpleFixedSizeExemplarReservoir extends FixedSizeExemplarReservoirBase {
```

Comment:
I find the behavior of this class slightly confusing. Is it documented anywhere? I think I see what it does, but it's not obvious why.

Reply:
It's documented [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#:~:text=SimpleExemplarReservoir%20This%20Exemplar%20reservoir%20MAY%20take%20a%20configuration%20parameter%20for%20the%20size%20of%20the%20reservoir%20pool.%20The%20reservoir%20will%20accept%20measurements%20using%20an%20equivalent%20of%20the%20naive%20reservoir%20sampling%20algorithm). Are you referring to sampling logic? The spec cites [this](https://en.wikipedia.org/wiki/Reservoir_sampling#Simple_algorithm) reservoir sampling algorithm on wikipedia

Reply:
According to that wikipedia article, the implementation is incorrect. The first `n` elements should be kept, filling the reservoir, and only after that should the random number be generated and checked. In our implementation, the random number is generated on every input regardless of if the reservoir is full yet or not.

Reply:
https://github.com/open-telemetry/opentelemetry-specification/issues/2205